forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,429 @@
+        Ok(())
+    }
+}
+
+struct NodeInfoOutput {
+    chain_status: ColoredString,
+    network: ColoredString,
+    uptime: String,
+    health: ColoredString,
+    wallet_address: ColoredString,
+}
+
+fn chain_status(node_status: &NodeStatusInfo) -> anyhow::Result<String> {
+    let NodeStatusInfo {
+        behind,
+        epoch,
+        base_fee,
+        sync_status,
+        ..
+    } = node_status;
+    let base_fee = format_balance_string(base_fee.clone(), FormattingMode::NotExactNotFixed)?;
+    let behind = {
+        let behind = Duration::from_secs(*behind);
+        let human_duration = humantime::format_duration(behind);
+        format!("{}", human_duration)
+    };
+
+    Ok(format!(
+        "[sync: {sync_status}! ({behind} behind)] [basefee: {base_fee}] [epoch: {epoch}]"
+    ))
+}
+
+fn fmt_info(
+    node_status: &NodeStatusInfo,
+    start_time: DateTime<Local>,
+    network: &str,
+    default_wallet_address: &Option<String>,
+    color: &LoggingColor,
+) -> anyhow::Result<NodeInfoOutput> {
+    let NodeStatusInfo { health, .. } = node_status;
+
+    let use_color = color.coloring_enabled();
+    let uptime_duration = Local::now() - start_time;
+    let uptime = {
+        format!(
+            "{}h {}m {}s (Started at: {})",
+            uptime_duration.num_hours(),
+            uptime_duration.num_minutes(),
+            uptime_duration.num_seconds(),
+            start_time
+        )
+    };
+
+    let chain_status = {
+        let status = chain_status(node_status)?;
+        if use_color {
+            status.blue()
+        } else {
+            status.normal()
+        }
+    };
+
+    let network = if use_color {
+        network.green()
+    } else {
+        network.normal()
+    };
+
+    let health = {
+        let s = format!("{health:.2}%\n\n");
+        if use_color {
+            if *health > 85. {
+                s.green()
+            } else {
+                s.red()
+            }
+        } else {
+            s.normal()
+        }
+    };
+
+    let default_wallet_address = {
+        let addr = default_wallet_address.clone().unwrap_or("-".to_string());
+        if use_color {
+            addr.bold()
+        } else {
+            addr.normal()
+        }
+    };
+
+    Ok(NodeInfoOutput {
+        chain_status,
+        network,
+        uptime,
+        health,
+        wallet_address: default_wallet_address,
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{
+        str::FromStr,
+        sync::Arc,
+        time::{Duration, SystemTime, UNIX_EPOCH},
+    };
+
+    use chrono::Local;
+    use colored::*;
+    use forest_blocks::{BlockHeader, Tipset};
+    use forest_cli_shared::logger::LoggingColor;
+    use forest_shim::{address::Address, econ::TokenAmount};
+    use fvm_shared::clock::EPOCH_DURATION_SECONDS;
+
+    use super::{fmt_info, get_node_status, NodeStatusInfo};
+    use crate::cli::info_cmd::{chain_status, SyncStatus};
+
+    fn mock_tipset_at(duration: u64) -> Arc<Tipset> {
```

Comment:
To clarify it, I'd name this argument `timestamp` or `seconds_since_unix_epoch`.

Reply:
done